### PR TITLE
Fixed tuple to List Conversion in Dataframe class

### LINKED ISSUE #25075: Segmentation fault while creating dataframe from list of mixed-type rows
```python
In [2]: pd.DataFrame([
   ...:     [1, 2],
   ...:     (3, 4)
   ...: ])
Segmentation fault: 11
```

I'd like to see a `TypeError` here, instead of a segfault.

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.2.final.0
python-bits: 64
OS: Darwin
OS-release: 18.2.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.0
pytest: None
pip: 19.0.1
setuptools: 40.7.1
Cython: None
numpy: 1.15.4
scipy: 1.2.0
pyarrow: None
xarray: None
IPython: 7.2.0
sphinx: None
patsy: None
dateutil: 2.7.5
pytz: 2018.9
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by Vibhu-Agarwal**
- [X] closes #25075 
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

Traceback of issue #25075 
- [pandas/core/frame.py: L#435](https://github.com/pandas-dev/pandas/blob/master/pandas/core/frame.py#L435) `to_arrays(data, columns, dtype=dtype)`
- [pandas/core/internals/construction.py: L#403](https://github.com/pandas-dev/pandas/blob/master/pandas/core/internals/construction.py#L403) `_list_to_arrays(data, columns, coerce_float=coerce_float, dtype=dtype)`
- [pandas/core/internals/construction.py: L#434](https://github.com/pandas-dev/pandas/blob/master/pandas/core/internals/construction.py#L434) `lib.to_object_array(data)`
- [pandas/_libs/lib.pyx: L#2283](https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/lib.pyx#L2283) `row[j]`

### Cause of "Segmentation fault"
**row** variable as declared on [line#2266](https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/lib.pyx#L2266) has to be of _list_ type.
It is tried to assign a tuple by converting to a list on [line#2280](https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/lib.pyx#L2280) using `row = <list>input_rows[i]`
The conversion fails. As a result, the type of **row** is tuple (not a list) after line#2280. The `Segmentation fault` is returned when tried to access a value from it at [line#2283](https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/lib.pyx#L2283)

### Fixes in #25089 
- At [line#2280](https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/lib.pyx#L2280) in [_libs/lib.pyx](https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/lib.pyx), changed the style of converting tuple into list.
  `<list>` --> `list()`
  This successfully converts tuples into list and further exception does not occur. As a result, function `to_object_array` correctly returns the numpy ndarray, what it's supposed to return.

As a result, not only `Segmentation fault` is removed, the need of function to raise `Type Error` is also eliminated.

### CODE REVIEW

**Comment by alimcmaster1** [comment]
Hey thanks for the PR - mind writing a simple test to check this raises a Type Error?

**Review by jreback — CHANGES_REQUESTED** [review]
-  needs a test (put in tests/frame/test_constructor.py)
- whatsnew for 0.24.2
- pls do an asv check

**Comment by Vibhu-Agarwal** [comment]
>     * needs a test (put in tests/frame/test_constructor.py)
> 
>     * whatsnew for 0.24.2
> 
>     * pls do an asv check

@jreback Will be adding the tests soon.
I'm not exactly aware about the **asv check** though.
Can you please explain me what it is or just point me to a good link.

**Comment by jreback** [comment]
http://pandas.pydata.org/pandas-docs/stable/development/contributing.html#running-the-performance-test-suite

**Review by jreback — APPROVED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]
waiting on tests / whatsnew / asv check

**Comment by Vibhu-Agarwal** [comment]
- [X] added tests
- [X] modified whatsnew
- [X] performed asv check

@jreback Made all the requested changes.

File: doc/source/whatsnew/v0.24.2.rst [hunk 1]
```diff
@@ -22,13 +22,14 @@ Fixed Regressions
 
 - Fixed regression in :meth:`DataFrame.all` and :meth:`DataFrame.any` where ``bool_only=True`` was ignored (:issue:`25101`)
 
+- Fixed regression in :meth:`DataFrame.__init__`  where mixed type of rows generated `Segmentation fault` (:issue:`25075`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.2.rst]: Fixed issue in ``DataFrame`` construction with passing a mixed list of mixed types could segfault.

File: pandas/tests/frame/test_constructors.py [hunk 2]
```diff
@@ -1183,6 +1183,13 @@ def test_constructor_mixed_dict_and_Series(self):
                              index=['a', 'b'])
         tm.assert_frame_equal(result, expected)
 
+    def test_constructor_mixed_type_rows(self):
+        data = [[1, 2], (3, 4)]
```
Comment (jreback) [inline pandas/tests/frame/test_constructors.py]: add the issue number as a comment

File: pandas/tests/frame/test_constructors.py [hunk 3]
```diff
@@ -1183,6 +1183,13 @@ def test_constructor_mixed_dict_and_Series(self):
...
+        result = DataFrame(data)
+        data2 = [[1, 2], [3, 4]]
```
Comment (jreback) [inline pandas/tests/frame/test_constructors.py]: you can inline data2 to the constructor

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by Vibhu-Agarwal** [comment]
@jreback Made all the requested changes and removed all unnecessary lines in whatsnew.

**Comment by jreback** [comment]
can you run the appropriate asvs for this

**Comment by Vibhu-Agarwal** [comment]
@jreback performed asv check with frame constructors (frame_ctor)
The report says `BENCHMARKS NOT SIGNIFICANTLY CHANGED.`

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @Vibhu-Agarwal

**Pull request merged**